src/App.jsx

```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_FIELD_BY_LENGTH = 'Sort by length';
+
+const sortGoods = (goods, field, isReversed) => {
+  const sortedGoods = [...goods];
+
+  switch (field) {
+    case SORT_FIELD_ALPHABETICALLY:
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SORT_FIELD_BY_LENGTH:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortedGoods = sortGoods(goodsFromServer, sortField, isReversed);
+  const isResetVisible = isReversed || sortField;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const isResetVisibleHandler = () => {
```

Comment:
```suggestion
  const handleReset = () => {
```
https://www.loom.com/share/fb0a4bee68d3479fb667555a7e702a15